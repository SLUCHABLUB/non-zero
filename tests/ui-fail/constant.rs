use non_zero::non_zero;

const _: std::num::NonZero<u8> = non_zero!(0);
const _: std::num::NonZero<u16> = non_zero!(0);
const _: std::num::NonZero<u32> = non_zero!(0);
const _: std::num::NonZero<u64> = non_zero!(0);
const _: std::num::NonZero<usize> = non_zero!(0);
const _: std::num::NonZero<i8> = non_zero!(0);
const _: std::num::NonZero<i16> = non_zero!(0);
const _: std::num::NonZero<i32> = non_zero!(0);
const _: std::num::NonZero<i64> = non_zero!(0);
const _: std::num::NonZero<isize> = non_zero!(0);

fn main() {}
