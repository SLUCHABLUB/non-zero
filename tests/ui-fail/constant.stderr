error[E0080]: evaluation of `_::{constant#0}` failed
 --> tests/ui-fail/constant.rs:3:34
  |
3 | const _: std::num::NonZero<u8> = non_zero!(0);
  |                                  ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:3:34
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/ui-fail/constant.rs:3:34
  |
3 | const _: std::num::NonZero<u8> = non_zero!(0);
  |                                  ^^^^^^^^^^^^
  |
  = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
 --> tests/ui-fail/constant.rs:4:35
  |
4 | const _: std::num::NonZero<u16> = non_zero!(0);
  |                                   ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:4:35
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/ui-fail/constant.rs:4:35
  |
4 | const _: std::num::NonZero<u16> = non_zero!(0);
  |                                   ^^^^^^^^^^^^
  |
  = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
 --> tests/ui-fail/constant.rs:5:35
  |
5 | const _: std::num::NonZero<u32> = non_zero!(0);
  |                                   ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:5:35
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/ui-fail/constant.rs:5:35
  |
5 | const _: std::num::NonZero<u32> = non_zero!(0);
  |                                   ^^^^^^^^^^^^
  |
  = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
 --> tests/ui-fail/constant.rs:6:35
  |
6 | const _: std::num::NonZero<u64> = non_zero!(0);
  |                                   ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:6:35
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/ui-fail/constant.rs:6:35
  |
6 | const _: std::num::NonZero<u64> = non_zero!(0);
  |                                   ^^^^^^^^^^^^
  |
  = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
 --> tests/ui-fail/constant.rs:7:37
  |
7 | const _: std::num::NonZero<usize> = non_zero!(0);
  |                                     ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:7:37
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/ui-fail/constant.rs:7:37
  |
7 | const _: std::num::NonZero<usize> = non_zero!(0);
  |                                     ^^^^^^^^^^^^
  |
  = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
 --> tests/ui-fail/constant.rs:8:34
  |
8 | const _: std::num::NonZero<i8> = non_zero!(0);
  |                                  ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:8:34
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/ui-fail/constant.rs:8:34
  |
8 | const _: std::num::NonZero<i8> = non_zero!(0);
  |                                  ^^^^^^^^^^^^
  |
  = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
 --> tests/ui-fail/constant.rs:9:35
  |
9 | const _: std::num::NonZero<i16> = non_zero!(0);
  |                                   ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:9:35
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/ui-fail/constant.rs:9:35
  |
9 | const _: std::num::NonZero<i16> = non_zero!(0);
  |                                   ^^^^^^^^^^^^
  |
  = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
  --> tests/ui-fail/constant.rs:10:35
   |
10 | const _: std::num::NonZero<i32> = non_zero!(0);
   |                                   ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:10:35
   |
   = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
  --> tests/ui-fail/constant.rs:10:35
   |
10 | const _: std::num::NonZero<i32> = non_zero!(0);
   |                                   ^^^^^^^^^^^^
   |
   = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
  --> tests/ui-fail/constant.rs:11:35
   |
11 | const _: std::num::NonZero<i64> = non_zero!(0);
   |                                   ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:11:35
   |
   = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
  --> tests/ui-fail/constant.rs:11:35
   |
11 | const _: std::num::NonZero<i64> = non_zero!(0);
   |                                   ^^^^^^^^^^^^
   |
   = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of `_::{constant#0}` failed
  --> tests/ui-fail/constant.rs:12:37
   |
12 | const _: std::num::NonZero<isize> = non_zero!(0);
   |                                     ^^^^^^^^^^^^ the evaluated program panicked at 'tried initialising a non-zero value to zero', $DIR/tests/ui-fail/constant.rs:12:37
   |
   = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
  --> tests/ui-fail/constant.rs:12:37
   |
12 | const _: std::num::NonZero<isize> = non_zero!(0);
   |                                     ^^^^^^^^^^^^
   |
   = note: this note originates in the macro `non_zero` (in Nightly builds, run with -Z macro-backtrace for more info)
